lib/chef/provider/powershell_script.rb

```diff
@@ -60,8 +60,8 @@ def command
       def flags
         interpreter_flags = [*default_interpreter_flags].join(" ")
 
-        if ! (new_resource.flags.nil?)
```

Comment:
It seems like we're making this much more complex than we need to. There may be a bigger issue here, but we should be able to do this all in the resource now that we're properly using the resource DSL in the script resource. In the powershell_script resource just add a flags resource with our desired default value and also perform the join via a proc.

Reply:
`[*default_interpreter_flags]` could be simplified to just `default_interpreter_flags` -- there ought to be a rubocop rule for that, because its pretty trivially redundant.

but even more broadly, it makes more sense to concatenate the arrays and then join at the end:

```
def flags
  flags = default_interpreter_flags
  flags += new_resource.flags unless new_resource.flags.nil?
  flags.join(" ")
end
```

so that's clearer about what is going on here. 

now the broader question -- yes, overriding this delegator via inheritance is a fairly terrible pattern.  moving it to the resource would be better but there needs to be a bit of caution here in the way that the default flags are concatenated and you want to make certain not to mutate the default value when you do that concatenation.

Reply:
Updated this logic. Using flags from `resources` and removed its entire manipulation from `provider`
Thanks for the review comment :)
Could you please check and let me know in case of any issues